hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -175,7 +175,8 @@ def set_element(self, data, result, idx, strategy=None):
                 "Generated array element %r from %r cannot be represented as "
                 "dtype %r - instead it becomes %r .  Consider using a more "
                 "precise strategy, as this will be an error in a future "
-                "version." % (val, strategy, self.dtype, result[idx])
+                "version." % (val, strategy, self.dtype, result[idx]),
+                since="2018-07-31",
```

Comment:
This deprecation was added on `2018-07-31`, but was then further tightened on `2018-09-22` for other data types (https://github.com/HypothesisWorks/hypothesis/commit/4b4bc41b31e08128966b58a605453ba1926d26af in #1592).

Reply:
I think we need to use the most recent date for this, since it changed the behavior and not just the message.  Sadly that means it will probably survive into v4 😢